src/parser/Parser.ts

```diff
@@ -1623,11 +1466,100 @@ export class Parser {
             },
             condition,
             thenBranch,
-            elseIfBranches,
-            elseBranch
+            elseBranch,
+            isInlineIfThen
         );
     }
-    private expressionStatement(expr: Expression, additionalTerminators: BlockTerminator[]): ExpressionStatement | IncrementStatement {
+
+    private tryParseBranch(ifToken: Token) {
+        //keep track of the current error count, because if the then branch fails,
+        //we will trash them in favor of a single error on if
+        let diagnosticsLengthBeforeBlock = this.diagnostics.length;
+
+        // we're parsing a multi-line ("block") form of the BrightScript if/then and must find
+        // a trailing "end if" or "else if"
+        let branch = this.block(TokenKind.EndIf, TokenKind.Else);
+
+        if (!branch) {
+            //throw out any new diagnostics created as a result of a `then` block parse failure.
+            //the block() function will discard the current line, so any discarded diagnostics will
+            //resurface if they are legitimate, and not a result of a malformed if statement
+            this.diagnostics.splice(diagnosticsLengthBeforeBlock, this.diagnostics.length - diagnosticsLengthBeforeBlock);
+
+            //this whole if statement is bogus...add error to the if token and hard-fail
+            this.diagnostics.push({
+                ...DiagnosticMessages.expectedEndIfElseIfOrElseToTerminateThenBlock(),
+                range: ifToken.range
+            });
+            throw this.lastDiagnosticAsError();
+        }
+        return branch;
+    }
+
+    private ensureNewLineOrColon(silent = false) {
+        const prev = this.previous().kind;
+        if (prev !== TokenKind.Newline && prev !== TokenKind.Colon) {
+            if (!silent) {
+                this.diagnostics.push({
+                    ...DiagnosticMessages.expectedNewlineOrColon(),
+                    range: this.peek().range
+                });
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private ensureInline(statements: Statement[]) {
+        statements.forEach(stat => {
+            if (isIfStatement(stat) && !stat.isInline) {
+                this.diagnostics.push({
+                    ...DiagnosticMessages.expectedInlineIfStatement(),
+                    range: stat.range
+                });
+            }
+        });
+    }
+
+    //consume then branch of inline if statement
+    private inlineBranch(...additionalTerminators: BlockTerminator[]): Block | undefined {
```

Comment:
If this is `if statement` specific, please give the method a more specific name.